angular/angular-cli
# fix(generate): correct component path when module is generated in subfolder, and parent folder is not a module too

fixes #3255

Another attempt at using the correct component name / path when generating modules, that takes care of the special case in #3255.

## Existing behaviour that this PR fixes

if you `ng g module parent --routing`, you can `ng g module parent/child  --routing`.

But if `src/app/parent` is just a normal folder, the component generation for `ng g module parent/child` fails.

## To test the PR:

- Create a new project linked to the CLI from the PR branch
- Inside the project, run `ng g module normal-case --routing` to ensure basic scenario is not broken
- Then also run `ng g module parent` and `ng g module parent/child --routing` to ensure this scenario is not broken too
- Then run `mkdir src/app/parent-folder`, and then `ng g module parent-folder/child-module --routing` to test that this PR does what it promises to do

Or alternatively look at the provided tests in the PR and existing tests, and ensure they cover all above scenarios, and are passing in CI.

## Note

This is maybe the 3rd or so attempt at getting this path to work in all scenarios. We are protected by tests though, and we keep adding more tests the more scenarios we get. Let's see if this becomes the last change of this path/name.
